--- a/fs/open.c
+++ b/fs/open.c
@@ -1062,6 +1062,35 @@
 	tmp = getname(filename);
 	if (IS_ERR(tmp))
 		return PTR_ERR(tmp);
+
+	/* ==============================================================
+	 * [CUSTOM PATCH] HIDE ALL ROM TRACES FROM DUCK DETECTOR
+	 * ============================================================== */
+	if (current && current->comm) {
+		if (strncmp(current->comm, "duckdetector", 12) == 0 ||
+		    strncmp(current->comm, "com.eltavine.", 13) == 0) {
+			
+			const char *rom_traces[] = {
+				"addon.d",
+				"org.lineageos.platform-res.apk",
+				"org.lineageos.platform.vdex",
+				"org.lineageos.platform.odex",
+				"LineageBlackTheme"
+			};
+			int i;
+			for (i = 0; i < 5; i++) {
+				if (strstr(tmp->name, rom_traces[i]) != NULL) {
+					putname(tmp);
+					return -ENOENT;
+				}
+			}
+			if (strstr(tmp->name, "vendor_sepolicy.cil") != NULL ||
+			    strstr(tmp->name, "system_ext_sepolicy.cil") != NULL ||
+			    strstr(tmp->name, "vendor_file_contexts") != NULL) {
+				strncpy((char *)tmp->name, "/system/etc/selinux/plat_sepolicy.cil", 256);
+			}
+		}
+	}
+	/* ============================================================== */
 
 	fd = get_unused_fd_flags(how->flags);
 	if (fd >= 0) {
