diff --git a/KernelSU/kernel/supercall/dispatch.c b/KernelSU/kernel/supercall/dispatch.c
--- a/KernelSU/kernel/supercall/dispatch.c
+++ b/KernelSU/kernel/supercall/dispatch.c
@@ -7,4 +7,5 @@
 #include <linux/version.h>
 #ifdef CONFIG_KSU_SUSFS
 #include <linux/susfs_def.h>
+#include <linux/susfs.h>
 #endif
@@ -339,7 +340,7 @@ static long ksu_handle_set_app_profile(void __user *arg)
     ret = ksu_set_app_profile(&cmd.profile);
     if (!ret) {
         ksu_persistent_allow_list();
-#ifdef CONFIG_KSU_TRACEPOINT_HOOK
+#if defined(CONFIG_KSU_TRACEPOINT_HOOK)
         ksu_mark_running_process();
 #endif
     }
@@ -630,12 +631,18 @@ long ksu_handle_supercall(struct pt_regs *regs)
         return 0;
     }
+#ifdef CONFIG_KSU_SUSFS
     case CMD_SUSFS_ADD_SUS_KSTAT: {
-        susfs_add_sus_kstat(arg);
-        return 0;
+        return susfs_add_sus_kstat(arg);
     }
+#ifdef CONFIG_KSU_SUSFS_SUS_KSTAT_REDIRECT
+    case CMD_SUSFS_ADD_SUS_KSTAT_REDIRECT: {
+        return susfs_add_sus_kstat_redirect(arg);
+    }
+#endif
     case CMD_SUSFS_UPDATE_SUS_KSTAT: {
-        susfs_update_sus_kstat(arg);
-        return 0;
+        return susfs_update_sus_kstat(arg);
     }
+#endif
     default:
         return -EINVAL;
     }
