diff --git a/KernelSU/kernel/supercall/dispatch.c b/KernelSU/kernel/supercall/dispatch.c
--- a/KernelSU/kernel/supercall/dispatch.c
+++ b/KernelSU/kernel/supercall/dispatch.c
@@ -7,2 +7,3 @@
 #ifdef CONFIG_KSU_SUSFS
 #include <linux/susfs_def.h>
+#include <linux/susfs.h>
 #endif
@@ -339,5 +340,5 @@ static long ksu_handle_set_app_profile(void __user *arg)
     if (!ret) {
         ksu_persistent_allow_list();
-#ifdef CONFIG_KSU_TRACEPOINT_HOOK
+#if defined(CONFIG_KSU_TRACEPOINT_HOOK)
         ksu_mark_running_process();
@@ -632,2 +633,7 @@ long ksu_handle_supercall(struct pt_regs *regs)
     case CMD_SUSFS_ADD_SUS_KSTAT: {
         return susfs_add_sus_kstat(arg);
     }
+#ifdef CONFIG_KSU_SUSFS_SUS_KSTAT_REDIRECT
+    case CMD_SUSFS_ADD_SUS_KSTAT_REDIRECT: {
+        return susfs_add_sus_kstat_redirect(arg);
+    }
+#endif
     case CMD_SUSFS_UPDATE_SUS_KSTAT: {
