--- a/KernelSU/kernel/supercall/dispatch.c
+++ b/KernelSU/kernel/supercall/dispatch.c
@@ -7,5 +7,6 @@
 #include <linux/version.h>
 #ifdef CONFIG_KSU_SUSFS
 #include <linux/susfs_def.h>
+#include <linux/susfs.h>
 #endif
 
@@ -339,7 +340,7 @@
     ret = ksu_set_app_profile(&cmd.profile);
     if (!ret) {
         ksu_persistent_allow_list();
-#ifdef CONFIG_KSU_TRACEPOINT_HOOK
+#if defined(CONFIG_KSU_TRACEPOINT_HOOK) && defined(CONFIG_KSU_SUSFS)
         ksu_mark_running_process();
 #endif
     }
@@ -632,6 +633,12 @@
     case CMD_SUSFS_ADD_SUS_KSTAT: {
         susfs_add_sus_kstat(arg);
         return 0;
     }
+#ifdef CONFIG_KSU_SUSFS_SUS_KSTAT_REDIRECT
+    case CMD_SUSFS_ADD_SUS_KSTAT_REDIRECT: {
+        susfs_add_sus_kstat_redirect(arg);
+        return 0;
+    }
+#endif
     case CMD_SUSFS_UPDATE_SUS_KSTAT: {
         susfs_update_sus_kstat(arg);
         return 0;
