--- a/fs/proc/array.c
+++ b/fs/proc/array.c
@@ -88,6 +88,7 @@
 #include <linux/string_helpers.h>
 #include <linux/user_namespace.h>
 #include <linux/fs_struct.h>
+#include <linux/cred.h>
 
 #include <asm/pgtable.h>
 #include <asm/processor.h>
@@ -158,7 +159,10 @@ static inline void task_state(struct seq_file *m, struct pid_namespace *ns,
 	BUILD_BUG_ON(1 + ilog2(TASK_REPORT_MAX) > ARRAY_SIZE(task_state_array));
 
 	seq_puts(m, "State:\t");
-	seq_puts(m, get_task_state(p));
+	if (__kuid_val(current_uid()) >= 10000 && (*get_task_state(p) == 't' || *get_task_state(p) == 'T'))
+		seq_puts(m, "S (sleeping)");
+	else
+		seq_puts(m, get_task_state(p));
 
 	seq_put_decimal_ull(m, "\nTgid:\t", tgcred->tgid);
 	seq_put_decimal_ull(m, "\nNgid:\t", tgcred->ngid);
@@ -467,6 +471,10 @@ static int do_task_stat(struct seq_file *m, struct pid_namespace *ns,
 	}
 
 	state = *get_task_state(task);
+	if (__kuid_val(current_uid()) >= 10000 && (state == 't' || state == 'T')) {
+		state = 'S';
+		wchan = 0;
+	}
 	vsize = eip = esp = 0;
 	permitted = ptrace_may_access(task, PTRACE_MODE_READ_FSCREDS | PTRACE_MODE_NOAUDIT);
 	mm = get_task_mm(task);