diff --git a/fs/proc/array.c b/fs/proc/array.c
index f5da74234..6a91e931f 100644
--- a/fs/proc/array.c
+++ b/fs/proc/array.c
@@ -179,6 +179,9 @@ static inline void task_state(struct seq_file *m, struct pid_namespace *ns,
 	task_unlock(p);
 	rcu_read_unlock();
 
+	if (current_uid().val >= 10000)
+		tpid = 0;
+
 	if (umask >= 0)
 		seq_printf(m, "Umask:\t%#04o\n", umask);
 	seq_puts(m, "State:\t");
diff --git a/fs/proc/cmdline.c b/fs/proc/cmdline.c
index fa762c5fb..3108c4a03 100644
--- a/fs/proc/cmdline.c
+++ b/fs/proc/cmdline.c
@@ -1,3 +1,5 @@
+#include <linux/slab.h>
+#include <linux/string.h>
 // SPDX-License-Identifier: GPL-2.0
 #include <linux/fs.h>
 #include <linux/init.h>
@@ -6,7 +8,23 @@
 
 static int cmdline_proc_show(struct seq_file *m, void *v)
 {
-	seq_puts(m, saved_command_line);
+	char *copy = kstrdup(saved_command_line, GFP_KERNEL);
+	char *ptr;
+	if (copy) {
+		if ((ptr = strstr(copy, "androidboot.verifiedbootstate=orange"))) strncpy(ptr + 30, "green ", 6);
+		if ((ptr = strstr(copy, "androidboot.verifiedbootstate=yellow"))) strncpy(ptr + 30, "green ", 6);
+		if ((ptr = strstr(copy, "androidboot.vbmeta.device_state=unlocked"))) strncpy(ptr + 32, "locked  ", 8);
+		if ((ptr = strstr(copy, "androidboot.flash.locked=0"))) strncpy(ptr + 25, "1", 1);
+		if ((ptr = strstr(copy, "androidboot.selinux=permissive"))) strncpy(ptr + 20, "enforcing ", 10);
+		if ((ptr = strstr(copy, "buildvariant=userdebug"))) strncpy(ptr + 13, "user     ", 9);
+		if ((ptr = strstr(copy, "buildvariant=eng"))) strncpy(ptr + 13, "user", 4);
+		if ((ptr = strstr(copy, "androidboot.mode=custom"))) strncpy(ptr + 17, "normal", 6);
+		if ((ptr = strstr(copy, "androidboot.secureboot=0"))) strncpy(ptr + 23, "1", 1);
+		if ((ptr = strstr(copy, "androidboot.lockstate=unlocked"))) strncpy(ptr + 22, "locked  ", 8);
+		seq_puts(m, copy);
+		kfree(copy);
+	} else {
+	}
 	seq_putc(m, '\n');
 	return 0;
 }
diff --git a/kernel/kallsyms.c b/kernel/kallsyms.c
index 126d4433d..fb02d5951 100644
--- a/kernel/kallsyms.c
+++ b/kernel/kallsyms.c
@@ -756,6 +756,9 @@ static int kallsyms_open(struct inode *inode, struct file *file)
 		return -ENOMEM;
 		reset_iter(iter, 0);
 
+	if (current_uid().val >= 10000)
+		return -ENOENT;
+
 	/*
 	 * Instead of checking this on every s_show() call, cache
 	 * the result here at open time.
diff --git a/security/selinux/selinuxfs.c b/security/selinux/selinuxfs.c
index 7415f49a3..49973e1a4 100644
--- a/security/selinux/selinuxfs.c
+++ b/security/selinux/selinuxfs.c
@@ -1,3 +1,5 @@
+#include <linux/sched.h>
+#include <linux/cred.h>
 // SPDX-License-Identifier: GPL-2.0-only
 /* Updated: Karl MacMillan <kmacmillan@tresys.com>
  *
